packages/components/src/interfaces/native/Prose.ts

```diff
@@ -8,33 +8,45 @@ import { ParagraphSchema } from "./Paragraph"
 import { TableSchema } from "./Table"
 import { UnorderedListSchema } from "./UnorderedList"
 
-export const ProseSchema = Type.Object(
-  {
-    type: Type.Literal("prose"),
-    content: Type.Optional(
-      Type.Array(
-        Type.Union([
-          Type.Ref(DividerSchema),
-          Type.Ref(HeadingSchema),
-          Type.Ref(OrderedListSchema),
-          Type.Ref(ParagraphSchema),
-          Type.Ref(TableSchema),
-          Type.Ref(UnorderedListSchema),
-        ]),
-        {
-          title: "Content block",
-          description: "A collection of native content components.",
-          minItems: 1,
-        },
+const BASE_PROSE_META = {
+
+  title: "Content block",
+  description: "A collection of native content components.",
+  format: "prose",
+
+}
+
+const generateProseSchema = (id?: string) => {
```

Comment:
this is done to generate the `ProseSchema` so that there's 1 with an `id` and 1 without. These are **both the same** in terms of validation.

however, the output for `compile` and to `JSONForms` is different - if we use a `Ref`, the schema that `JSONForms` sees **is just the ref** and causes the prose editor to not be rendered.

cc @dcshzj if i missed anything

Reply:
also if build relies on the `id` implicitly

Reply:
Hmm curious, what if we removed the Type.Ref from the getComponentSchema inside main.ts? That usage means that among all the component schemas, only the prose schema is the one being Type.Ref'd.

https://github.com/opengovsg/isomer/blob/e000312c83dd69d19b5fc1f884adad09dd25b748/packages/components/src/schemas/main.ts#L30

Reply:
Discussed offline, this wouldn't work cos we still get duplicate $id, one from the actual schema inside `getComponentSchema` and another one from the definitions.